src/main/java/edu/umich/ctools/sectionsUtilityTool/FriendServlet.java

```diff
@@ -0,0 +1,207 @@
+		String pathInfo = request.getPathInfo();
+		PrintWriter out = response.getWriter();
+		JsonObject json;
+		Friend myFriend = new Friend();
+		switch (pathInfo){
+		case "/friendCreate":
+			String friendInviteEmail = request.getParameter("id");
+			String friendInstructorEmail = request.getParameter("inst_email");
+			String friendInstructorFirstName = request.getParameter("inst_first_name");
+			String friendInstructorLastName = request.getParameter("inst_last_name");
+			json = doFriendCreate(myFriend, 
+					friendInviteEmail, 
+					friendInstructorEmail, 
+					friendInstructorFirstName, 
+					friendInstructorLastName);
+			break;
+		default:
+			json = Json.createObjectBuilder()
+					.add("message", "request error")
+					.build();
+			break;
+		}
+		out.print(json.toString());
+		out.flush();
+	}
+	
+	private JsonObject doFriendCreate(Friend myFriend, 
+			String inviteEmail, 
+			String instructorEmail, 
+			String instructorFirstName, 
+			String instructorLastName){
+		JsonObject json;
+		String instructorName = instructorFirstName + " " + instructorLastName;
+		String detailedMessage;
+		myFriend.init();
+		//Step 1: determine if friend exists
+		int friendExists = Friend.checkAccountExist(inviteEmail);
+		switch(friendExists){
+		case -1:
+			//If friend email is invalid, send message for invalid email in json response.
+			M_log.warn(" friendCreate: invalid email address " + inviteEmail);
+			detailedMessage = "Account " + inviteEmail + " is not a valid email address";
+			json = Json.createObjectBuilder()
+					.add("message", "false")
+					.add("detailedMessage", detailedMessage)
+					.build();
+			break;
+		case 0:
+			//If friend has valid email but does not have a friend account, attempt to create one.
+			M_log.info(" friendCreate: " + inviteEmail + " does not have an account. One will attempt to be created.");
+			int friendCreate = Friend.doSendInvite(inviteEmail, instructorEmail, instructorName);
+			//Step 2: attempt to create friend account
+			switch(friendCreate){
+			case 1:
+				//If friend account is successful, send json response back saying so
+				M_log.info(" friendCreate: successfully created account for  " + inviteEmail);
+				detailedMessage = "Friend Account created for " + inviteEmail;
+				json = Json.createObjectBuilder()
+						.add("message", "true")
+						.add("detailedMessage", detailedMessage)
+						.build();
+				
+				//Step 3: send notification to instructor indicating that a friend account has been created.
+				Friend.notifyCurrentUser(instructorName, instructorEmail, inviteEmail);
+				break;
+			case -1:
+				//If attempt to create friend account fails, then send so in response
+				M_log.warn(" friendCreate: invalid email address " + inviteEmail);
+				detailedMessage = "Friend Account NOT created for " + inviteEmail;
+				json = Json.createObjectBuilder()
+						.add("message", "false")
+						.add("detailedMessage", detailedMessage)
+						.build();
+				break;
+			default:
```

Comment:
I see that `Json.createObjectBuilder` is being repeated in switch cases, Might well make a tiny function out of it